icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -5889,6 +5890,13 @@ void RBBITest::TestBug22602() {
     RuleBasedBreakIterator bi(rule, pe, ec);
 }
 
+void RBBITest::TestBug22636() {
+    UParseError pe {};
+    UErrorCode ec {U_ZERO_ERROR};
+    RuleBasedBreakIterator bi(u"A{77777777777777}", pe, ec);
```

Comment:
It would be good to check that 2147483647 is accepted and that 2147483648 fails, just to verify that the check is doing the right thing on the boundary.

Reply:
DONE. also add Java fix